tests/testthat/setup-testing_depth.R

```diff
@@ -0,0 +1,57 @@
+#' Returns testing depth set by an environmental variable.
+#'
+#' @details Looks for the option `TESTING_DEPTH` first, if not set,
+#' takes the system environmental variable `TESTING_DEPTH`. If neither
+#' is set, then returns 3 by default. If the value of `TESTING_DEPTH`
+#' is not a scalar numeric, then returns 3.
+#'
+#' @return `numeric(1)` the testing depth.
+#'
+get_testing_depth <- function() {
+  default_depth <- 3
+  depth <- getOption("TESTING_DEPTH", default_depth)
+  if (is.null(depth)) {
+    depth <- Sys.getenv("TESTING_DEPTH", default_depth)
+  }
```

Comment:
this way you will never have `is.null(depth)` hence you will never use `Sys.getenv`

```r
getOption("XSSASASACASX", 2)
#> 2
```

Reply:
What about 

```r
depth <- getOption("TESTING_DEPTH", Sys.getenv("TESTING_DEPTH"))
```

Reply:
We can, if the user sets the depth as "bazinga!"

Reply:
Yes, this is perfect.

```r
default_depth <- 3
depth <- getOption("TESTING_DEPTH", Sys.getenv("TESTING_DEPTH", default_depth))
```

Reply:
nice!

Just remember `Sys.getenv` returns characters

```r
> Sys.getenv("TESTING_DEPTH", default_depth)
[1] "3"
> default_depth
[1] 3
```

Reply:
but later `as.numeric` should be fine

Reply:
Yes sir